src/sentry/integrations/slack/link_team.py

```diff
@@ -0,0 +1,206 @@
+        integration_id=integration.id,
+        slack_id=slack_id,
+        channel_id=channel_id,
+        channel_name=channel_name,
+        response_url=response_url,
+    )
+
+    return absolute_uri(
+        reverse("sentry-integration-slack-link-team", kwargs={"signed_params": signed_params})
+    )
+
+
+class SelectTeamForm(forms.Form):
+    team = forms.ChoiceField(label="Team")
+
+    def __init__(self, teams, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+
+        self.fields["team"].choices = [(team.id, team.slug) for team in teams]
+        self.fields["team"].widget.choices = self.fields["team"].choices
+
+
+class SlackLinkTeamView(BaseView):
+    def get_identity(self, integration, slack_id):
+        try:
+            idp = IdentityProvider.objects.get(type="slack", external_id=integration.external_id)
+        except IdentityProvider.DoesNotExist:
+            logger.error(
+                "slack.action.invalid-team-id", extra={"slack_id": integration.external_id}
+            )
+            return self.respond(status=403)
+
+        try:
+            identity = Identity.objects.select_related("user").get(idp=idp, external_id=slack_id)
+        except Identity.DoesNotExist:
+            # I don't think this could be possible but just in case
+            return self.respond(status=403)
+        return identity
+
+    def send_slack_message(self, request, client, token, text, channel_id, integration):
+        payload = {
+            "token": token,
+            "channel": channel_id,
+            "text": text["body"],
+        }
+        headers = {"Authorization": "Bearer %s" % token}
+        try:
+            client.post("/chat.postMessage", headers=headers, data=payload, json=True)
+        except ApiError as e:
+            message = str(e)
+            if message != "Expired url":
+                logger.error("slack.link-notify.response-error", extra={"error": message})
+        else:
+            return render_to_response(
+                "sentry/slack-post-linked-team.html",
+                request=request,
+                context={
+                    "heading_text": text["heading"],
+                    "body_text": text["body"],
+                    "channel_id": channel_id,
+                    "team_id": integration.external_id,
+                },
+            )
+
+    @transaction_start("SlackLinkTeamView")
+    @never_cache
+    def handle(self, request, signed_params):
+        params = unsign(signed_params)
+        integration = Integration.objects.get(id=params["integration_id"])
+        teams = Team.objects.filter(
+            organization__in=integration.organizations.all(), status=TeamStatus.VISIBLE
+        ).order_by("slug")
+        channel_name = params["channel_name"]
+        channel_id = params["channel_id"]
+        form = SelectTeamForm(teams, request.POST or None)
+        if request.method == "POST" and not form.is_valid():
+            return Response(status=422)
```

Comment:
If the form is not valid then we should return a "400 Bad Request" instead of a 422.